mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerConnector.java

```diff
@@ -152,10 +153,15 @@ protected String getStratifiedQuery(String tableName, int limit, String strata,
   @Override
   protected Schema getSchema(int sqlType, String typeName, int scale, int precision, String columnName,
                              boolean isSigned, boolean handleAsDecimal) throws SQLException {
-    if (SqlServerSourceSchemaReader.shouldConvertToDatetime(typeName)) {
+    if (Types.TIMESTAMP == sqlType) {
+      // DateTime, DateTime2 and SmallDateTime cases
       return Schema.of(Schema.LogicalType.DATETIME);
     }
 
+    if (SqlServerSourceSchemaReader.DATETIME_OFFSET_TYPE == sqlType) {
+      return Schema.of(Schema.LogicalType.TIMESTAMP_MICROS);
+    }
+
     if (SqlServerSourceSchemaReader.GEOMETRY_TYPE == sqlType || SqlServerSourceSchemaReader.GEOGRAPHY_TYPE == sqlType) {
```

Comment:
How are we validating this ?

Reply:
This method is not being called so removed it altogether